source/plugins/go/input/lib/cadvisor.go

```diff
@@ -0,0 +1,1219 @@
+	INSIGHTSMETRICS_TAGS_GPU_MODEL         = "gpuModel"
+	INSIGHTSMETRICS_TAGS_GPU_ID            = "gpuId"
+	INSIGHTSMETRICS_TAGS_PV_NAMESPACE      = "container.azm.ms/pv"
+	INSIGHTSMETRICS_TAGS_PVC_NAME          = "pvcName"
+	INSIGHTSMETRICS_TAGS_PVC_NAMESPACE     = "pvcNamespace"
+	INSIGHTSMETRICS_TAGS_POD_NAME          = "podName"
+	INSIGHTSMETRICS_TAGS_PV_CAPACITY_BYTES = "pvCapacityBytes"
+	INSIGHTSMETRICS_TAGS_VOLUME_NAME       = "volumeName"
+	INSIGHTSMETRICS_FLUENT_TAG             = "oms.api.InsightsMetrics"
+	INSIGHTSMETRICS_TAGS_POD_UID           = "podUid"
+	PV_KUBE_SYSTEM_METRICS_ENABLED_EVENT   = "CollectPVKubeSystemMetricsEnabled"
+)
+
+var (
+	telemetryCpuMetricTimeTracker           = time.Now().Unix()
+	telemetryMemoryMetricTimeTracker        = time.Now().Unix()
+	telemetryPVKubeSystemMetricsTimeTracker = time.Now().Unix()
+	nodeTelemetryTimeTracker                = map[string]interface{}{}
+	operatingSystem                         string
+	nodeCpuUsageNanoSecondsLast             float64
+	nodeCpuUsageNanoSecondsTimeLast         string
+	linuxNodePrevMetricRate                 float64
+	winNodePrevMetricRate                   = map[string]float64{}
+	winNodeCpuUsageNanoSecondsLast          = map[string]float64{}
+	winNodeCpuUsageNanoSecondsTimeLast      = map[string]interface{}{}
+	Log 					*logrus.Logger
+)
+
+func init() {
+	Log = logrus.New()
+
+	osType := os.Getenv("OS_TYPE")
+	if osType != "" && osType == "windows" {
+		LogPath = WindowsLogPath + "kubernetes_perf_log.txt"
+	} else {
+		LogPath = LinuxLogPath + "kubernetes_perf_log.txt"
+	}
+
+	isTestEnv := os.Getenv("ISTEST") == "true"
+	if isTestEnv {
+		LogPath = "./kubernetes_perf_log.txt"
+	}
+
+	// Open file
+	file, err := os.OpenFile(LogPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if err != nil {
+		Log.Fatalf("Failed to open log file: %v", err)
+	}
+
+	Log.SetOutput(file)
+	Log.SetLevel(logrus.InfoLevel)
+}
+
+type metricDataItem map[string]interface{}
+
+func getSummaryStatsFromCAdvisor(winNode map[string]string) (*http.Response, error) {
+	relativeUri := "/stats/summary"
+	return getResponse(winNode, relativeUri)
+}
+
+// func getCongifzCAdvisor(winNode map[string]string) (*http.Response, error) {
+// 	relativeUri := "/configz"
+// 	return getResponse(winNode, relativeUri)
+// }
+
+// func getAllMetricsCAdvisor(winNode map[string]string) (*http.Response, error) {
+// 	relativeUri := "/metrics/cadvisor"
+// 	return getResponse(winNode, relativeUri)
+// }
+
+func GetPodsFromCAdvisor(winNode map[string]string) (*http.Response, error) {
+	relativeUri := "/pods"
+	return getResponse(winNode, relativeUri)
+}
+
+func getBaseCAdvisorUrl(winNode map[string]string) string {
+	cAdvisorSecurePort := isCAdvisorOnSecurePort()
+
+	var defaultHost string
+	if cAdvisorSecurePort {
+		defaultHost = fmt.Sprintf("https://localhost:%s", CADVISOR_SECURE_PORT)
+	} else {
+		defaultHost = fmt.Sprintf("http://localhost:%s", CADVISOR_NON_SECURE_PORT)
```

Comment:
FYI.. https://localhost:[10250|10255] is useless code and probable we can get rid of this to address devskim alert?

Reply:
not sure I understand